# HG changeset patch
# User Rik <rik@octave.org>
# Date 1385441833 28800
#      Mon Nov 25 20:57:13 2013 -0800
# Branch stable
# Node ID dc58d4810de611028bea586c886d3d4deb5e6e3a
# Parent  13c0c762fae185534997c5d6e0b4b05423171e63
main.in.cc: Respect --silent option and don't print startup message.

* main.in.cc (main): Pass "--silent" option and friends on to octave
executable rather than filtering it out.

diff --git a/src/main.in.cc b/src/main.in.cc
--- a/src/main.in.cc
+++ b/src/main.in.cc
@@ -631,18 +631,22 @@ main (int argc, char **argv)
           // If we see this option, then we can just exec octave; we
           // don't have to create a child process and wait for it to
           // exit.  But do exec "octave", not "octave-cli", because even
           // if the --no-gui option is given, we may be asked to do some
           // plotting or ui* calls.
 
           // This option calls the cli executable for the 3.8 release.
         }
-      else if (! strcmp (argv[i], "--silent") || ! strcmp (argv[i], "-q"))
-        warn_display = false;
+      else if (! strcmp (argv[i], "--silent") || ! strcmp (argv[i], "-q")
+               || ! strcmp (argv[i], "--quiet"))
+        {
+          warn_display = false;
+          new_argv[k++] = argv[i];
+        }
       else
         new_argv[k++] = argv[i];
     }
 
   new_argv[k] = 0;
 
   if (gui_libs || start_gui)
     {
